src/test-links.ts

```diff
@@ -86,12 +88,18 @@ export class MockLink extends ApolloLink {
       throw new Error(`Mocked response should contain either result or error: ${key}`);
     }
 
-    return new Observable<FetchResult>(observer => {
+    return new Observable(observer => {
       let timer = setTimeout(() => {
         if (error) {
           observer.error(error);
         } else {
-          if (result) observer.next(result);
+          if (result) {
+            observer.next(
+              Object.prototype.toString.call(result) === '[object Function]'
```

Comment:
```suggestion
              typeof result === 'function'
```
should be perfectly safe here.